ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -653,7 +657,7 @@ MiResolveDemandZeroFault(IN PVOID Address,
     ASSERT(PointerPte->u.Hard.Valid == 0);
 
     /* Assert we have enough pages */
-    ASSERT(MmAvailablePages >= 32);
+    //ASSERT(MmAvailablePages >= 32);
```

Comment:
If we correctly counted available pages, the check below is eqivalent to `MmAvailablePages == 0`. And yes at some point we might run out of pages. I thought about bailing out on 32, but the "problem" is that we only have one way to allocate pages anyway (well, 2, but they behave the same), so even if we "reserved" 32 pages, we don't have a path that can allocate them, so why reserve them in the first place? If we have code that requires a certain amount of pages no matter what, even at IRQL >= DISPATCH_LEVEL, then we should fix that by pre-allocating the pages instead of expecting the PFN database to respect other components needs by randomly reserving an arbitrary amount of pages. I haven't seen that happening though.